fix: Document {{auto}} properly [ISSUE-737] (#4544)
getsentry/sentry-docs: src/platforms/common/enriching-events/identify-user.mdx

Users consist of a few critical pieces of information that construct a unique identity in Sentry. Each of these is optional, but one **must** be present for the Sentry SDK to capture the user:

<DefinitionList>

<PlatformSection notSupported={["apple"]}>

### `id`

Your internal identifier for the user.
</PlatformSection>

<PlatformSection supported={["apple"]}>

### `userId`

Your internal identifier for the user.
</PlatformSection>

### `username`

The username. Typically used as a better label than the internal id.

### `email`

An alternative, or addition, to the username. Sentry is aware of email addresses and can display things such as Gravatars and unlock messaging capabilities.

### `ip_address`

The user's IP address. If the user is unauthenticated, Sentry uses the IP address as a unique identifier for the user. Sentry will attempt to pull this from the HTTP request data (`request.env.REMOTE_ADDR` field in JSON), if available. That might require <PlatformIdentifier name="send-default-pii" /> set to `true` in the SDK options.

If the user's `ip_address` is set to `"{{auto}}"`, Sentry will infer the IP address from the connection between your app and Sentry's server.

> If the field is omitted, the default value is `null`. However, due to backwards compatibility concerns, certain platforms (in particular JavaScript) have a different default value of `"{{auto}}"`. SDKs and other clients are strongly discouraged from relying on this behavior and should set IP-addresses or `"{{auto}}"` explicitly.

Reviewer: We can't really say that SDKs and other clients are strongly discouraged because it's giving them human qualities. We can either say the makers of those SDKs and other clients are strongly discouraged or perhaps the suggestion I've made below. Whichever you prefer.

```suggestion
If the field is omitted, the default value is `null`. However, due to backwards compatibility concerns, certain platforms (in particular JavaScript) have a different default value for `"{{auto}}"`. SDKs and other clients should not rely on this behavior and should set IP addresses or `"{{auto}}"` explicitly.
```

Author: yup thanks, your suggestion makes sense. i think programmers tend to anthropomorphize their creations so it's normal to me.

> To opt-out of collecting users' IP addresses, you can use Sentry's [server-side data](/product/data-management-settings/scrubbing/) scrubbing to remove `$user.ip_address`. Adding such a rule ultimately overrules any other logic.

Reviewer: ```suggestion
To opt out of collecting users' IP addresses, you can use Sentry's [server-side data](/product/data-management-settings/scrubbing/) scrubbing to remove `$user.ip_address`. Adding such a rule ultimately overrules any other logic.
```

`ip_address`

: The user's IP address. If the user is unauthenticated, Sentry uses the IP address as a unique identifier for the user. Sentry will attempt to pull this from the HTTP request data (`request.env.REMOTE_ADDR` field in JSON), if available. That might require <PlatformIdentifier name="send-default-pii" /> set to `true` in the SDK options.

> If the user's `ip_address` is set to `"{{auto}}"`, Sentry will infer the IP address from the connection between your app and Sentry's server. For certain SDKs, particularly iOS and frontend JavaScript, this is the default behavior and requires no further configuration. For most other SDKs the default is not to collect any IP address. If you find that you're missing the user's IP address, setting this value to `"{{auto}}"` explicitly might help.

Reviewer 2: this paragraph should belong under `ip_address`, but it looks like "breaks out" of that section and is de-indented:

<img width="782" alt="Screenshot 2022-01-03 at 11 55 45 AM" src="https://user-images.githubusercontent.com/5597345/147957813-d1b2c639-9945-4a3b-bb51-5b5f87b416fc.png">

i think this could cause a bit of confusion - is there a way to fix this?

Reviewer: Indenting in markdown without showing anything isn't always very straightforward in Markdown, but I wonder if the `: ` would work on the following paragraphs.

Author: I found a component in our docs page that works like the `: `, but allows for paragraphs.

> If the user's `ip_address` is set to `"{{auto}}"`, Sentry will infer the IP address from the connection between your app and Sentry's server. For certain SDKs, particularly iOS and frontend JavaScript, this is the default behavior and requires no further configuration. For most other SDKs the default is not to collect any IP address. If you find that you're missing the user's IP address, setting this value to `"{{auto}}"` explicitly might help.

Reviewer: ```suggestion
If the user's `ip_address` is set to `"{{auto}}"`, Sentry will infer the IP address from the connection between your app and Sentry's server. For certain SDKs, particularly iOS and frontend JavaScript, this is the default behavior and requires no further configuration. For most other SDKs the default is to not collect any IP address. If you find that you're missing the user's IP address, setting this value to `"{{auto}}"` explicitly might help.
```

<!--
SDK devs who read this and wonder why their SDK does not behave like the JS
SDK: Relay changes the default value from `null` to `{{auto}}` for the
platforms javascript, cocoa, objc. If you want to collect IP addresses by
default because you're building a frontend/desktop SDK, just send {{auto}} by

> default. We don't really want to special-case more platforms.

Reviewer 2: nit: are developers usually referencing sentry docs in a way that would make comments like these visible to them? using the native documents as an example (https://sentry-docs-git-doc-ip-auto.sentry.dev/platforms/native/enriching-events/identify-user/) it's not immediately obvious to me how i would find this bit of information without directly referencing the source document. this looks like useful information, but it's squirreled away in a place that doesn't appear to be too discoverable for those that it would help.

Author: i now went with less technical phrasing such that it can be shown in user docs

See https://github.com/getsentry/relay/blob/54fc479259c5c8a70c48ad1994d5e8eb8629ba32/relay-general/src/store/normalize.rs#L380-L383
-->

> To opt-out from collecting user's IP addresses, you can use Sentry's [server-side data](/product/data-management-settings/scrubbing/) scrubbing to remove `$user.ip_address`. Adding such a rule ultimately overrules any other logic.

Reviewer: ```suggestion
To opt-out of collecting users' IP addresses, you can use Sentry's [server-side data](/product/data-management-settings/scrubbing/) scrubbing to remove `$user.ip_address`. Adding such a rule ultimately overrules any other logic.
```
